java/common/src/main/java/org/ray/util/MethodId.java

```diff
@@ -0,0 +1,224 @@
+package org.ray.util;
+
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.ConcurrentHashMap;
+import org.objectweb.asm.Type;
+import org.ray.util.logger.RayLog;
+
+
+public final class MethodId {
+
+  /**
+   * SerializedLambda.implMethodKind
+   * REF_NONE                    = 0
+   * REF_getField                = 1,
+   * REF_getStatic               = 2,
+   * REF_putField                = 3,
+   * REF_putStatic               = 4,
+   * REF_invokeVirtual           = 5,
+   * REF_invokeStatic            = 6,
+   * REF_invokeSpecial           = 7,
+   * REF_newInvokeSpecial        = 8,
+   * REF_invokeInterface         = 9,
+   * REF_LIMIT                  = 10.
+   */
+
+  private static final ConcurrentHashMap<String, MethodId> MAP = new ConcurrentHashMap<>(256);
+  /**
+   * format A.B.C.cname
+   */
+  public final String className;
+  public final String methodName;
+  public final String methodDesc;
```

Comment:
`fromMethod` and `fromSerializedLambda` use 2 different ways to generate `methodDesc`.
I think it should be okay because their formats are the same, and not likely to change in the future.
But if it changes, it could cause some problems that are hard to debug.
I think we can 
1) add a note here to remind this potential issue.
2) add a unit test that compares the 2 formats. So if future versions generate different results, we'll quickly know it.